### ISSUE TITLE
default time zones of input data

### ISSUE THREAD

**Issue by kmike** [scrapinghub/dateparser#230]
Hey! Why does dateparser convert date *to* timezone specified in settings, not *from* this timezone? 

For example, let's say there is a forum with UTC+5 timezone, so all dates are in UTC+5, and I want to have dates in UTC. Passing TIMEZONE='UTC+5' won't help - dateparser will assume dates are in UTC, and then convert them to UTC+5, i.e. it'll add five hours instead of subtracting five hours.

Once you have a date in a known timezone it is easy to convert it to any other timezone using other libraries, you don't need dateparser support for this. So currently 'TIMEZONE' option is merely a helper. 

The problem is that it is tricky to handle relative dates correctly with dateparser now. Let's say we're on that forum again, with UTC+5 dates. Old dates are written in full, e.g. "2016-09-07 12:30", recent dates are relative, e.g. "1 hour ago".

Without timezone information dateparser will think both dates are in UTC. It means it'll use utcnow() as a current time when computing a full date for "1 hour ago", which is not what they meant on a website.

Let's say current time (as used by website) is 2016-09-07 15:00 UTC+5, or 2016-09-07 10:00 UTC.

text  | date in UTC+5 | date in UTC | parse result
-----|---------------|------------- | ------------
1 hour ago | 2016-09-07 14:00 | 2016-09-07 09:00 | 2016-09-07 09:00 UTC
2016-09-07 12:30 | 2016-09-07 12:30 | 2016-09-07 07:30 | 2016-09-07 12:30 UTC

As you see, 'parse result' is incorrect. TIMEZONE option doesn't help here because both times are adjusted by the same amount.

`RELATIVE_BASE` can help, but the code is far from obvious:

```py
from datetime import datetime
import pytz

forum_tz = pytz.timezone("UTC+5")  # forum time zone

# current time as tz-aware UTC datetime
now_utc = pytz.utc.localize(datetime(2016, 9, 7, 10)) 

# here goes a hack: even though docs say 
# "Dateparser assumes all dates to be in UTC",
# we have to pass RELATIVE_BASE as a naive (not tz-aware) 
# datetime in UTC+5 timezone
forum_now = now_utc.astimezone(forum_tz)  # current time in forum timezone
settings = {
   'RELATIVE_BASE': forum_now.replace(tzinfo=None),
}
dt = dateparser.parse(date_string, settings=settings)

# the returned dt is not in UTC, it is in UTC+5. Convert it back to UTC:
dt = forum_tz.localize(dt).astimezone(pytz.utc)
```

What do you think about adding DEFAULT_INPUT_TIMEZONE option to handle this use case? Maybe it also worths it to rename 'TIMEZONE' option to 'RESULT_TIMEZONE', in order to avoid confusion.

**Comment by waqasshabbir** [comment]
Hi @kmike! thank you for the detailed description. `TIMEZONE` setting has been causing confusion to lots of users as it can be easily misconstrued for supplied date timezone. We never provided a mechanism to specify a default input timezone for dates. The only _relatively_ straightforward way to accomplish that is by appending the timezone next to date string and define the result timezone for the same.

```
parse('2016-09-07 12:30 UTC+5', settings={'RELATIVE_BASE': datetime(2016, 9, 7, 15, 0), 'TIMEZONE': 'UTC+5'})
datetime.datetime(2016, 9, 7, 12, 30)
```

But above doesn't work for relative dates and must be fixed for both reasons a) it's not straightforward b) it's not all inclusive.

I'll add/rename the settings. Will be pushing a fix soon.

**Comment by waqasshabbir** [comment]
fixed by #235
